converter/internal/common/validate.go

```diff
@@ -10,9 +10,17 @@ import (
 )
 
 func UnsupportedNotDeepEquals(a any, b any, name string) diag.Diagnostics {
+	return UnsupportedNotDeepEqualsMessage(a, b, name, "")
+}
+
+func UnsupportedNotDeepEqualsMessage(a any, b any, name string, message string) diag.Diagnostics {
 	var diags diag.Diagnostics
 	if !reflect.DeepEqual(a, b) {
-		diags.Add(diag.SeverityLevelError, fmt.Sprintf("unsupported %s config was provided.", name))
+		if message != "" {
+			diags.Add(diag.SeverityLevelError, fmt.Sprintf("unsupported %s config was provided. %s", name, message))
```

Comment:
Nesting messages with `:` instead of periods is more common in Go IME

```suggestion
			diags.Add(diag.SeverityLevelError, fmt.Sprintf("unsupported %s config was provided: %s", name, message))
```

Reply:
all set - I implemented this outside of the suggestion since a test needed to be updated at the same time.